client/src/modules/patients/visits/modals/visits.modal.html

```diff
@@ -0,0 +1,209 @@
+    </p>
+    <div class="radio">
+      <label>
+        <input 
+          type="radio" 
+          name="inside_health_zone" 
+          id="yes" 
+          ng-value="1" 
+          ng-model="AdmitCtrl.visit.inside_health_zone"
+          ng-change="AdmitCtrl.onChangeHZ()" checked>
+        <span translate>PATIENT_RECORDS.VISITS.IS_IN_HEALTH_ZONE</span>
+      </label>
+    </div>
+    <div class="radio">
+      <label>
+        <input 
+          type="radio" 
+          name="inside_health_zone" 
+          id="no" 
+          ng-value="0" 
+          ng-model="AdmitCtrl.visit.inside_health_zone"
+          ng-change="AdmitCtrl.onChangeHZ()">
+        <span translate>PATIENT_RECORDS.VISITS.IS_NOT_IN_HEALTH_ZONE</span>
+      </label>
+    </div>
+  </div>
+
+  <br>
+
+  <!-- only on discharge -->
+  <div ng-if="!AdmitCtrl.isAdmission">
+    <bh-discharge-select on-change-callback="AdmitCtrl.onSelectDischarge(dischargeType)"></bh-discharge-select>
+  </div>
+
+  <div
+    class="form-group"
+    ng-class="{ 'has-error' : VisitForm.$submitted && VisitForm.diagnosis.$invalid }">
+
+    <label class="control-label"><span translate>FORM.LABELS.DIAGNOSIS</span></label>
+
+    <p class="text-info"><i class="fa fa-info-circle"></i> <span translate>PATIENT_RECORDS.VISITS.SEARCH_INFO</span></p>
+    <input
+      type="text"
+      name="diagnosis"
+      class="form-control"
+      ng-model="AdmitCtrl.visit.diagnosis"
+      uib-typeahead="diagnosis as diagnosis.label for diagnosis in AdmitCtrl.diagnoses | filter:$viewValue | limitTo:8"
+      typeahead-on-select="AdmitCtrl.select()"
+      autocomplete="off"
+      typeahead-min-length="3"
+      ng-required="AdmitCtrl.REQUIRE_DIAGNOSES">
+    </label>
+
+    <div class="help-block" ng-messages="VisitForm.diagnosis.$error" ng-show="VisitForm.$submitted">
+      <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
+    </div>
+  </div>
+
+  <div
+    class="form-group"
+    ng-class="{ 'has-error' : VisitForm.$submitted && VisitForm.notes.$invalid }">
+    <label class="control-label"><span translate>FORM.LABELS.NOTES</span></label>
+    <textarea name="notes" ng-model="AdmitCtrl.visit.notes" class="form-control"></textarea>
+    <div class="help-block">
+      <p translate>PATIENT_RECORDS.VISITS.NOTES_INFO</p>
+    </div>
+
+    <div class="help-block" ng-messages="VisitForm.notes.$error" ng-show="VisitForm.$submitted">
+      <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
+    </div>
+  </div>
+  
+  <!-- ward and bed -->
+  <div ng-if="AdmitCtrl.isAdmission && AdmitCtrl.visit.hospitalized">
+    <hr>
+    <bh-room-assign on-select-callback="AdmitCtrl.onBedRoomSelect(bed)"></bh-room-assign>
+  </div>
+</div>
+
+<div class="modal-footer">
+  <button type="button" class="btn btn-default" ng-click="AdmitCtrl.cancel()" data-method="cancel">
+    <span translate>FORM.BUTTONS.CLOSE</span>
+  </button>
+
+  <bh-loading-button loading-state="AdmitCtrl.$loading">
```

Comment:
Out of curiosity, why did you choose to manage `$loading` yourself instead of using `bh-submit`?